### ISSUE TITLE
Sync up and review encoding changes for multipart/form-data

### ISSUE THREAD

**Issue by travisleithead** [w3c/html#222]
Filing an issue to track the ongoing bugilla bug: https://www.w3.org/Bugs/Public/show_bug.cgi?id=18135

Conversation continued in the WHATWG cloned bug: https://www.w3.org/Bugs/Public/show_bug.cgi?id=16909, where a recent PR was suggested as potentially fixing the issue: https://github.com/whatwg/html/pull/710.

The relevant section is: http://w3c.github.io/html/sec-forms.html#multipart-form-data and the PR to WHATWG has **not** yet been merged into the W3C html spec.

Feedback was waiting on @masinter.

CC @domenic for tracking on his end

**Comment by masinter** [comment]
not sure what is wanted here, pls explain

**Comment by travisleithead** [comment]
I believe the request is to read the linked section and see if you are satisfied with the way it links RFC 7578.

> I believe this might have been fixed in https://github.com/whatwg/html/pull/710. **Larry, could you confirm that https://html.spec.whatwg.org/#multipart/form-data-encoding-algorithm correctly delegates to RFC 7578, and that RFC 7578 handles the cases discussed [here](https://www.w3.org/Bugs/Public/show_bug.cgi?id=16909)?**
> 
> From my reading I am not so sure... the HTML spec now says "Encode the (now mutated) form data set using the rules described by RFC 7578", but I can't find an algorithm in RFC 7578 that takes as input a form data set and gives as output a byte stream.

**Comment by masinter** [comment]
I'm not particularly satisfied with RFC 7578, but I couldn't justify holding back on the draft after it went through last call. There is no specified algorithm for taking a form data set and producing a byte stream.
The RFC makes requirements on the result, .and I think writing _some_ algorithm might be a good idea.

As it is, RFC 7578 gives requirements for the results but leaves some things open.

What I would suggest is completing the test harness for generating forms, filling, submitting, then checking the submitted  data, then testing current browser behavior. 

My (unconfirmed) belief is that currently browsers vary widely. If so there is still a choice to declare one the winner, or to leave the implementation less tightly constrained.

**Comment by masinter** [comment]
https://github.com/masinter/multipart-form-data 
has the beginning of a test framework.

**Comment by chaals** [comment]
We now reference RFC 7578, which seems to approximate best practice. There may be some value in defining a custom encoding algorithm, but it probably makes more sense just to update the reference if a better one comes along.

**Comment by masinter** [comment]
issues or PRs on https://github.com/masinter/multipart-form-data would be preferable to waiting for a better spec to arise spontaneously.
